core/store/src/trie/mem/resharding.rs

```diff
@@ -140,19 +138,17 @@ impl<'a, M: ArenaMemory> MemTrieUpdate<'a, M> {
                 let child_key = [key_nibbles, extension_nibbles].concat();
                 self.retain_multi_range_recursive(new_child_id, child_key, intervals_nibbles);
 
-                if self.updated_nodes[new_child_id] == Some(UpdatedMemTrieNode::Empty) {
-                    self.place_node(node_id, UpdatedMemTrieNode::Empty);
-                } else {
-                    self.place_node(
-                        node_id,
-                        UpdatedMemTrieNode::Extension {
-                            extension,
-                            child: OldOrUpdatedNodeId::Updated(new_child_id),
-                        },
-                    );
-                }
+                self.place_node(
+                    node_id,
+                    UpdatedMemTrieNode::Extension {
+                        extension,
+                        child: OldOrUpdatedNodeId::Updated(new_child_id),
+                    },
+                );
             }
         }
+
+        self.squash_node(node_id);
```

Comment:
The correctness of squashing depends on it being called on nodes in the correct order. 

sanity check: Is it called in the right order here? 

nit: Can you add a comment about it for future us?